smart-contracts/contracts/CosmosBridge.sol

```diff
@@ -179,35 +271,46 @@ contract CosmosBridge is CosmosBridgeStorage, Oracle {
         require(sigs.length == claims.length, "INV_CLM_LEN");
         require(sigs.length == signatureData.length, "INV_SIG_LEN");
 
+        uint256 intermediateNonce = lastNonceSubmitted;
+        lastNonceSubmitted += claims.length;
+
         for (uint256 i = 0; i < sigs.length; i++) {
+            require(intermediateNonce + 1 + i == claims[i].nonce, "INV_ORD");
             _submitProphecyClaimAggregatedSigs(sigs[i], claims[i], signatureData[i]);
         }
     }
 
+    /**
+     * @notice Submits a ProphecyClaim
+     * @param hashDigest The hashed message
+     * @param claimData The claim itself
+     * @param signatureData The signature data
+     */
     function submitProphecyClaimAggregatedSigs(
         bytes32 hashDigest,
         ClaimData calldata claimData,
         SignatureData[] calldata signatureData
     ) external {
+        uint256 previousNonce = lastNonceSubmitted;
+        require(previousNonce + 1 == claimData.nonce, "INV_ORD");
+
+        // update the nonce
+        lastNonceSubmitted = claimData.nonce;
```

Comment:
Let's create a test where an ERC20 token tries to pull a reentrancy attack on this code and it receives an `INV_ORD` error upon reentry.